apic_ml2/neutron/db/l3out_vlan_allocation.py

```diff
@@ -0,0 +1,241 @@
+
+            # process vlan ranges for each configured l3out network
+            for (l3out_network,
+                 vlan_range) in self.l3out_vlan_ranges.items():
+                # determine current configured allocatable vlans for
+                # this l3out network
+                vlan_ids = set()
+                vlan_min, vlan_max = vlan_range
+                vlan_ids |= set(moves.xrange(vlan_min, vlan_max + 1))
+
+                # remove from table unallocated vlans not currently
+                # allocatable
+                if l3out_network in allocations:
+                    for alloc in allocations[l3out_network]:
+                        try:
+                            # see if vlan is allocatable
+                            vlan_ids.remove(alloc.vlan_id)
+                        except KeyError:
+                            # it's not allocatable, so check if its allocated
+                            if not alloc.allocated:
+                                # it's not, so remove it from table
+                                LOG.debug("Removing vlan %(vlan_id)s on "
+                                          "l3out network "
+                                          "%(l3out_network)s from pool",
+                                          {'vlan_id': alloc.vlan_id,
+                                           'l3out_network':
+                                           l3out_network})
+                                self.session.delete(alloc)
+                    del allocations[l3out_network]
+
+                # add missing allocatable vlans to table
+                for vlan_id in sorted(vlan_ids):
+                    alloc = L3OutVlanAllocation(l3out_network=l3out_network,
+                                                vrf='',
+                                                vlan_id=vlan_id,
+                                                allocated=False)
+                    self.session.add(alloc)
+
+            # remove from table unallocated vlans for any unconfigured
+            # l3out networks
+            for allocs in allocations.itervalues():
+                for alloc in allocs:
+                    if not alloc.allocated:
+                        LOG.debug("Removing vlan %(vlan_id)s on l3out "
+                                  "network %(l3out_network)s from pool",
+                                  {'vlan_id': alloc.vlan_id,
+                                   'l3out_network':
+                                   alloc.l3out_network})
+                        self.session.delete(alloc)
+
+    def get_type(self):
+        return p_const.TYPE_VLAN
+
+    def reserve_vlan(self, l3out_network, vrf):
+        with self.session.begin(subtransactions=True):
+            query = (self.session.query(L3OutVlanAllocation).
+                     filter_by(l3out_network=l3out_network,
+                               vrf=vrf))
+            count = query.update({"allocated": True})
```

Comment:
question: is 'count' the number of rows updated? Is so, can you make that clear in the log message?